src/common/ant.ts

```diff
@@ -87,6 +91,99 @@ export class ANT {
    */
   static fork = forkANT;
 
+  /**
+   * Upgrade an ANT by forking it to the latest version and reassigning names.
+   *
+   *
+   * @param config Configuration object for the upgrade process
+   * @returns Promise resolving to the forked process ID and successfully reassigned names
+   */
+  static async upgrade({
+    signer,
+    antProcessId,
+    names,
+    arioProcessId = ARIO_MAINNET_PROCESS_ID,
+    ao,
+    logger = Logger.default,
+    antRegistryId,
+    skipVersionCheck = false,
+    antProcess = ANT.init({
+      process: new AOProcess({ processId: antProcessId, ao }),
+      signer,
+    }),
+    onSigningProgress,
+  }: {
+    signer: AoSigner;
+    antProcessId: string;
+    antProcess?: AoANTRead;
+    names: string[];
+    arioProcessId?: string;
+    skipVersionCheck?: boolean;
+    ao?: AoClient;
+    logger?: Logger;
+    antRegistryId?: string;
+    onSigningProgress?: (
+      name: keyof SpawnAntProgressEvent | 'reassigning-name',
+      payload:
+        | SpawnAntProgressEvent[keyof SpawnAntProgressEvent]
+        | { name: string },
+    ) => void;
+  }): Promise<{
+    forkedProcessId: string;
+    reassignedNames: string[];
+    failedReassignedNames: string[];
+  }> {
+    if (!skipVersionCheck) {
+      const currentVersion = await antProcess.getVersion();
+      const latestVersion = await ANT.versions.getLatestANTVersion();
+      if (currentVersion === latestVersion.version) {
+        return {
+          forkedProcessId: antProcessId,
+          reassignedNames: [],
+          failedReassignedNames: [],
+        };
+      }
+    }
+
+    const forkedProcessId = await ANT.fork({
+      signer,
+      antProcessId,
+      ao,
+      logger,
+      antRegistryId,
+      onSigningProgress,
+    });
+
+    // Step 2: Create a writable instance of the original ANT for reassigning names
+    const writableAnt = ANT.init({
+      process: new AOProcess({ processId: antProcessId, ao, logger }),
+      signer,
+    });
+
+    // we could parallelize this, but then signing progress would be harder to track
+    const reassignedNames: string[] = [];
+    const failedReassignedNames: string[] = [];
+    for (const name of names) {
+      try {
+        onSigningProgress?.('reassigning-name', { name });
+
+        await writableAnt.reassignName({
+          name,
+          arioProcessId,
+          antProcessId: forkedProcessId,
+        });
+
+        reassignedNames.push(name);
+      } catch (error) {
+        logger.error(`Failed to reassign name ${name}:`, { error });
+        // Continue with other names rather than failing completely
+        failedReassignedNames.push(name);
+      }
+    }
+
+    return { forkedProcessId, reassignedNames, failedReassignedNames };
```

Comment:
could be nice to return newAntProcess - essentially clones the current one with the configurations used including signer, ao, etc. 

If remaining with a static function this would use the provided params to construct it, else it would use the configured class properties.